### ISSUE TITLE
SpecifyObjectiveGradient is not an option for fmincon in R2015b

### ISSUE THREAD

**Issue by dweindl** [ICB-DCM/PESTO#29]
PestoOptions.m:37

Same for specificyConstraintGradient

**Comment by paulstapor** [comment]
I had the vague feeling that something like this was the case... Unfortunately, the Mathworks homepage claims this option would be compatible to R2013a....

**Comment by dweindl** [comment]
and:

No appropriate method, property, or field 'MaxIterations' for class 'optim.options.Fmincon'.

in getMultiStarts.m:185

**Comment by dweindl** [comment]
How important ist R2015 compatibility?

Just got the e-mail annoucing prerelease 2017a...

**Comment by FFroehlich** [comment]
I would aim for R2015 compatibility, even for us the newest version of matlab on our interactive servers is R2015b, on the SVM its even R2014

**Comment by paulstapor** [comment]
Well, in the first, there's just a typo, it should be "SpecifyObjectiveGradient", but the second option is definitely correct. Unfortunately, the MathWorks HP claims, this would be compatible to R2013a

**Comment by paulstapor** [comment]
Yep, Caro uses for example R2015... We should aim at R2014-compatibility

**Comment by FFroehlich** [comment]
mathworks does not claim it is backwards-compatible to R2013a as far as I can see, just that it was introduced in R2013a!

**Comment by dweindl** [comment]
And extroduced in R2015 again? :)

Well then, R2014 or R2015?

**Comment by dweindl** [comment]
related: 
```
Error in getMultiStarts (line 203)
        if (~options.fmincon.SpecifyObjectiveGradient)
```

**Comment by paulstapor** [comment]
Maybe it's a better decision to switch things back to optimset, although optimoptions is recommended for fmincon. In the former version, we used optimset and there were no problems like these...

**Comment by FFroehlich** [comment]
I would aim for R2014b, anything before that is not reasonable anyways due to the changes in the graphics engine

**Comment by paulstapor** [comment]
If we take optimset, the properties have different names and "MaxIterations" is "MaxIter", which is also known to older Matlab versions. Then, we should have compatibility down to R2014b... I'm just hesitating, as actually optimoptions with its specific Properties is recommended...

**Comment by dweindl** [comment]
So I guess it is
1) use optimset
or
2) or check for the current release to choose the option names
Although somewhat stupid to change to the deprecated option, (1) would be the easiest...

**Comment by paulstapor** [comment]
Ok, I'll do that

**Comment by FFroehlich** [comment]
"that" means (1)?

**Comment by dweindl** [comment]
Fixed by @paulstapor in afa58b8dc6cd156c290ff19e9bab449cc362b05c .

Successfully tested enzymatic catalysis, conversion reaction and transfection examples on R2015b.
